### PR TITLE
fix(oauth): Add some random expiration time to access tokens

### PR DESCRIPTION
**Pull request by vbudhram**
Fixes https://github.com/mozilla/fxa/issues/2608

Started working on this with Jared. It will add some configurable random amount of expiration time to tokens.

@shane-tomlinson @vladikoff Thoughts on the approach?

### CODE REVIEW

File: packages/fxa-auth-server/fxa-oauth-server/lib/db/mysql/index.js [hunk 1]
```diff
@@ -469,15 +472,17 @@ MysqlStore.prototype = {
     return this._write(QUERY_CODE_DELETE, [hash]);
   },
   generateAccessToken: function generateAccessToken(vals) {
+    // In certain scenarios to help avoid thundering herd, we add some padding to the expiration time of the token.
+    vals.ttl = (vals.ttl * 1000 || MAX_TTL) + EXPIRES_AT_TTL_PADDING;
```
Comment (6a68) [inline packages/fxa-auth-server/fxa-oauth-server/lib/db/mysql/index.js]: Should the second term in the sum be something like 

`(Math.floor(Math.random() * EXPIRES_AT_TTL_PADDING))`?

Comment (vbudhram) [reply packages/fxa-auth-server/fxa-oauth-server/lib/db/mysql/index.js]: Missed the most important part 🙇🏽‍♂️, fixed!

File: packages/fxa-auth-server/fxa-oauth-server/lib/db/mysql/index.js [hunk 2]
```diff
@@ -469,15 +472,19 @@ MysqlStore.prototype = {
     return this._write(QUERY_CODE_DELETE, [hash]);
   },
   generateAccessToken: function generateAccessToken(vals) {
+    // In certain scenarios to help avoid thundering herd, we add some padding to the expiration time of the token.
+    vals.ttl =
```
Comment (lmorchard) [inline packages/fxa-auth-server/fxa-oauth-server/lib/db/mysql/index.js]: Maybe not a real problem and just a nit, but modifying incoming params in a function always looks like future side-effect trouble to me. Might be better to assign to a new const var and use that below.

**Comment by 6a68** [comment]
Looks like this failed on the `cargo audit` errors addressed by #2622. Rebased to pick up that fix.

**Comment by lmorchard** [comment]
Might've rebased against the wrong commit? Looks like this is 164 commits now 😕

**Comment by 6a68** [comment]
yeah, I thought I force-pushed from the right branch back into this one, but I still see a ton of extra commits hanging out. I can clean this up in a bit.

**Comment by 6a68** [comment]
Looks like I've now fixed the bad rebase. Now this is train-145, plus the CI fix for the email server, and what I think is the latest version of the actual fix commit. Still might need to pull in the nsp fixes to get this to pass, we'll see.

**Comment by vbudhram** [comment]
Since this won't land today and train-146 gets out on Monday, it makes sense to target this against that train, closing here.

**Comment by clouserw** [comment]
@vbudhram you closed this here -- was there a PR somewhere else against a different train?

**Pull request closed without merging**